ambry-router/src/main/java/com.github.ambry.router/CryptoJobHandler.java

```diff
@@ -30,28 +34,54 @@
  * and any new jobs submitted after close will be ignored.
  */
 class CryptoJobHandler implements Closeable {
+  private final long cryptoJobTimeoutMs;
   private final AtomicBoolean enabled = new AtomicBoolean(false);
   private static final GeneralSecurityException CLOSED_EXCEPTION =
       new GeneralSecurityException("CryptoJobHandler closed");
+  private static final GeneralSecurityException CRYPTO_JOB_TIMEOUT_EXCEPTION =
+      new GeneralSecurityException("CryptoJob timeout", new TimeoutException());
   private ExecutorService scheduler;
+  private ConcurrentHashMap<CryptoJob, Long> pendingCryptoJobs = new ConcurrentHashMap<>();
+  private final Time time;
 
   private static final Logger logger = LoggerFactory.getLogger(CryptoJobHandler.class);
 
   /**
    * Instantiates {@link CryptoJobHandler}
+   * @param threadCount total number of worker threads
+   * @param cryptoJobTimeoutMs crypto job timeout in ms
+   * @param time {@link Time} instance to use
    */
-  CryptoJobHandler(int threadCount) {
+  CryptoJobHandler(int threadCount, long cryptoJobTimeoutMs, Time time) {
+    this.cryptoJobTimeoutMs = cryptoJobTimeoutMs;
     enabled.set(true);
     scheduler = Executors.newFixedThreadPool(threadCount);
+    this.time = time;
+  }
+
+  /**
+   * Cleans up expired crypto jobs
+   */
+  protected void cleanUpExpiredCryptoJobs() {
```

Comment:
over-riding it in some tests in PutManager so that timeouts tests are not affected by calls to cleanUpExpiredCryptoJobs() (since this method will move the mockTime)